pages/lists/[listId]/edit.tsx

```diff
@@ -207,133 +216,196 @@ export default function EditListPage({ list, initialContributors }: EditListPage
     );
   };
 
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+  const [deleteLoading, setDeleteLoading] = useState(false);
+
+  const handleOnDelete = () => {
+    setIsDeleteModalOpen(true);
+  };
+
+  const handleDeleteModalClose = () => {
+    setIsDeleteModalOpen(false);
+  };
+
+  const handleOnDeleteConfirm = async () => {
+    setSubmitted(true);
+    setDeleteLoading(true);
+
+    try {
+      const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/lists/${list.id}`, {
+        method: "DELETE",
+        headers: {
+          Authorization: `Bearer ${sessionToken}`,
+        },
+      });
+
+      if (res.ok) {
```

Comment:
refactor (blocking): use `fetchApiData`.


```suggestion
      const { error } = await fetchApiData({
        path: "/lists/${list.id}",
        method: "DELETE",
        bearerToken: sessionToken,
        pathValidator: () => true,
      });

      if (!error)
```